test/base/src/androidTest/kotlin/io/realm/kotlin/test/shared/MutableRealmIntTests.kt

```diff
@@ -0,0 +1,290 @@
+            equalityTest(c1, c2)
+        }
+    }
+
+    @Test
+    fun managed_nullability() {
+        realm.writeBlocking {
+            val c1 = copyToRealm(Sample())
+            nullabilityTest(c1)
+        }
+    }
+
+    @Test
+    fun managed_compareTo() {
+        realm.writeBlocking {
+            val c1 = copyToRealm(Sample())
+            val r1 = c1.mutableRealmIntField
+            r1.set(0)
+
+            val c2 = copyToRealm(Sample())
+            val r2 = c2.mutableRealmIntField
+            r2.set(Long.MAX_VALUE)
+            assertEquals(-1, r1.compareTo(r2))
+
+            r2.decrement(Long.MAX_VALUE)
+            assertEquals(0, r1.compareTo(r2))
+            r2.decrement(Long.MAX_VALUE)
+            assertEquals(1, r1.compareTo(r2))
+        }
+    }
+
+    @Test
+    fun managed_setOutsideTransactionThrows() {
+        val r = realm.writeBlocking {
+            copyToRealm(Sample())
+        }.mutableRealmIntField
+
+        assertFailsWithMessage<IllegalStateException>("Cannot set") {
+            r.set(22)
+        }
+    }
+
+    @Test
+    fun managed_incrementOutsideTransactionThrows() {
+        val r = realm.writeBlocking {
+            copyToRealm(Sample())
+        }.mutableRealmIntField
+
+        assertFailsWithMessage<IllegalStateException>("Cannot increment/decrement") {
+            r.increment(1)
+        }
+        assertFailsWithMessage<IllegalStateException>("Cannot increment/decrement") {
+            r.decrement(1)
+        }
+    }
+
+    @Test
+    fun managed_accessors() {
+        val r = MutableRealmInt.of(22)
+        realm.writeBlocking {
+            val sample = copyToRealm(Sample())
+            assertNotNull(sample.mutableRealmIntField)
+            sample.mutableRealmIntField = r
+            val managedMutableInt = sample.mutableRealmIntField
+            assertNotNull(managedMutableInt)
+            assertEquals(22, managedMutableInt.get())
+
+            assertNull(sample.nullableMutableRealmIntField)
+            sample.nullableMutableRealmIntField = r
+            val managedNullableMutableRealmInt = sample.nullableMutableRealmIntField
+            assertNotNull(managedNullableMutableRealmInt)
+            assertEquals(22, managedNullableMutableRealmInt.get())
+        }
+    }
+
+    @Test
+    fun managed_shareValueAcrossObjects() {
+        realm.writeBlocking {
+            val counter = MutableRealmInt.of(42)
+            val managedFoo = copyToRealm(Sample())
+            val managedBar = copyToRealm(Sample())
+
+            managedFoo.mutableRealmIntField = counter
+            managedBar.mutableRealmIntField = managedFoo.mutableRealmIntField
+            managedBar.mutableRealmIntField.increment(1)
+            val managedFooValue = managedFoo.mutableRealmIntField.get()
+            val managedBarValue = managedBar.mutableRealmIntField.get()
+
+            // Values obviously diverge since we don't copy the reference but the value, just as we
```

Comment:
We should probably document this behaviour in the Dokka for the class since this behaviour differs between unmanaged and managed objects.

Reply:
That's fine but it should be the same for all other primitive types. Should we add it elsewhere more visible? I find it a bit counterproductive to adding it only for `MutableRealmInt` and not making it visible for all other primitive types.